pkg/cloud/baremetal/actuators/machine/actuator_test.go

```diff
@@ -303,10 +327,49 @@ func TestChooseHost(t *testing.T) {
 					ProviderSpec: providerSpec5,
 				},
 			},
-			Hosts:            []runtime.Object{&host2, &host_with_label},
+			Hosts:            []runtime.Object{&host2, &hostWithLabel},
 			ExpectedHostName: "",
 			Config:           config5,
 		},
+		{
+			// No host chosen given only externally provisioned choices
+			Machine: machinev1beta1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Machine",
+					APIVersion: machinev1beta1.SchemeGroupVersion.String(),
+				},
+				Spec: machinev1beta1.MachineSpec{
+					ProviderSpec: providerSpec5,
+				},
+			},
+			Hosts:            []runtime.Object{&externallyProvisionedHost},
+			ExpectedHostName: "",
+			Config:           config,
+		},
+		{
+			// No host chosen given only externally provisioned choices
```

Comment:
this comment seems to be wrong, as we expect the how with the matched consumeRef to be chosen